### PR TITLE
Improve Output behavior.

### PR DESCRIPTION
**Pull request by Metalaka**
See https://github.com/hoaproject/Console/pull/55#discussion_r42047716

`Hoa\Console::getOutput` return `Hoa\Stream\IStream\Out` and then type
check must be done to allow correct uses of `Hoa\Console\Output`'s
multiplexer feature. Requiring type check in methods usage looks like
a wrong approach.
`Hoa\Console\Output` can embed an instance of `Hoa\Stream\IStream\Out`.
Like that `Output::isMultiplexerConsidered` &
`Output::considerMultiplexer` are always available. And when
`Output::write` is called, an internal call to `Hoa\Stream\IStream\Out`
is done.

### CODE REVIEW

File: Console.php [hunk 1]
```diff
@@ -323,12 +323,12 @@ public static function setOutput(Stream\IStream\Out $output)
...
      * Get output layer.
      *
-     * @return  \Hoa\Stream\IStream\Out
+     * @return  \Hoa\Console\Output
      */
     public static function getOutput()
     {
         if (null === static::$_output) {
-            static::$_output = new Output();
+            static::$_output = new Output(null);
```
Comment (Hywan) [inline Console.php]: `null` is not necessary here.

File: Output.php [hunk 2]
```diff
@@ -105,7 +126,9 @@ public function writeString($string)
     {
         $string = (string) $string;
 
-        return $this->write($string, strlen($string));
+        $this->write($string, strlen($string));
```
Comment (Hywan) [inline Output.php]: You can return the result of `$this->write`, this is still correct. Same comment for other similar modifications.

File: Window.php [hunk 3]
```diff
@@ -574,7 +566,7 @@ function () {
                 'hoa://Event/Console/Window:resize',
                 $_window,
                 new Core\Event\Bucket([
-                    'size' => \Window::getSize()
+                    'size' => Window::getSize()
```
Comment (Hywan) [inline Window.php]: Correct oO.

**Comment by Hywan** [comment]
I am actually doing the same thing with an `Input` class in https://github.com/hoaproject/Console/pull/51.

File: Output.php [hunk 4]
```diff
@@ -56,6 +56,25 @@ class Output implements Stream\IStream\Out
...
+
+
+    /**
+     * Constructor.
+     *
+     * @param  \Hoa\Stream\IStream\Out  $output    The output stream or NULL.
+     */
+    public function __construct(Stream\IStream\Out $output = null)
+    {
+        $this->_output = $output;
```
Comment (Hywan) [inline Output.php]: What about saying:
```php
use Hoa\File;

…

if (null === $output) {
    $output = new File\Write('php://stdout');
}
```

This is what I am doing in #51.

File: Output.php (hunk 2 above)
Comment (Metalaka) [inline Output.php]: Functionally correct yes, but it's more a style/logical issue;
* Returning somethings when there is `@return void` in the PHPDoc sound wrong.
* Using the return value of a void function seem also strange.

**Comment by Hywan** [comment]
Looks correct.

**Comment by Hywan** [comment]
Hmm, there is an issue with tests. We use `->output` to test the output. Unfortunately, it does not consider `php://stdout`, which is different for PHP. So, sorry but… I guess this is dangerous to work on `php://stdout` with `Hoa\File\Write` and using `echo` seems to be a good solution here. Consequently, we need to test `if (null === $this->_output)` everywhere to put some `echo`s everywhere.

Sorry for this undo!

**Comment by Hywan** [comment]
This is not an issue with testing only actually. I meant: `php://stdout` seems different than `echo` in this low-level. I don't know exactly why though.

**Comment by Metalaka** [comment]
Corrected.

**Comment by Hywan** [comment]
Seems correct!
